app/src/main/kotlin/model/abstractGraph/Graph.kt

```diff
@@ -73,4 +89,135 @@ abstract class Graph<D> {
     }
 
     abstract fun getEdge(vertex1: Vertex<D>, vertex2: Vertex<D>): Edge<D>
+
+    /* For every vertex, calculates normalized closeness centrality, based on which the key vertices are picked.
+     * Formula was taken from "Efficient Top-k Closeness Centrality Search" by Paul W. Olsen et al.,
+     * yet an easier algorithm for traversal was chosen. */
+    fun findKeyVertices(): Set<Vertex<D>> {
+        val graphSize = vertices.size
+
+        val distanceMap = getWeightMap()
+        if (!distanceMap.values.all { it >= 0 })
+            throw IllegalArgumentException("Cannot run key vertices finding algorithm for graph with negative edges.")
```

Comment:
I think it's too bad to throw an exception in that case

Reply:
I think you're right. I guess I'll replace it with `return null`, so that we can send a message to user that operation cannot be performed because of negative edges.